src/plugins/intel_npu/tools/multi-provider-inference-tool/tools/multi_provider_blobs_comparison.py

```diff
@@ -0,0 +1,305 @@
+            output_data, leaf_default_values
+        )
+        provider_result["outputs"].append(output_name)
+    return provider_result
+
+
+def get_comparison_profiver_output_files_result_data(
+    ref_provider_output_blob_file_data, provider_to_cmp_output_blob_file_data, datatype
+):
+    provider_to_cmp_output_blob_file_data["std_correlation"] = bin_diff.compare_blobs(
+        ref_provider_output_blob_file_data["path"],
+        provider_to_cmp_output_blob_file_data["path"],
+        datatype,
+    )
+    return provider_to_cmp_output_blob_file_data
+
+
+def get_comparison_provider_output_result_data(
+    ref_provider_name,
+    ref_provider_output_data,
+    provider_to_cmp_name,
+    provider_to_cmp_output_data,
+):
+    for field_to_compare in ["shape", "element_type"]:
+        if (
+            ref_provider_output_data[field_to_compare]
+            != provider_to_cmp_output_data[field_to_compare]
+        ):
+            provider_to_cmp_output_data["status"].append(
+                f'Incorrect "{field_to_compare}":'
+                f" {provider_to_cmp_output_data[field_to_compare]}, expected:"
+                f" {ref_provider_output_data[field_to_compare]}"
+            )
+            ref_provider_output_data["status"].append(
+                f'Incorrect "{field_to_compare}":'
+                f" {ref_provider_output_data[field_to_compare]}, expected:"
+                f" {provider_to_cmp_output_data[field_to_compare]}"
+            )
+    # If a reference output blob file is found in rhs provider,
+    # we will conduct compasion of these files.
+    # Otherwise, we do nothing.
+
+    # Eventually, if rhs provider has different outputs blob files than the reference provider,
+    # then the data cannot be compared to the reference data and it will holds NaN,
+    # which is respobisility of a consumer to decide how they will evaluate that
+
+    # Ultimately, we compare only outputs/files that are coherent and
+    # skip unmatching outputs/files
+    for f in ref_provider_output_data["files"]:
+        if f in provider_to_cmp_output_data["data"].keys():
+            try:
+                provider_to_cmp_output_data["data"][f] = (
+                    get_comparison_profiver_output_files_result_data(
+                        ref_provider_output_data["data"][f],
+                        provider_to_cmp_output_data["data"][f],
+                        ref_provider_output_data["element_type"],
+                    )
+                )
+            except RuntimeError as ex:
+                provider_to_cmp_output_data["status"].append(
+                    f"Cannot compare the file: {f}, err: {ex}"
+                )
+        else:
+            if "provider" not in provider_to_cmp_output_data["not_found_data"][f].keys():
+                provider_to_cmp_output_data["not_found_data"][f]["provider"] = []
+            provider_to_cmp_output_data["not_found_data"][f]["provider"].append(ref_provider_name)
```

Comment:
This pattern repeats multiple times:
```
if "provider" not in some_dict["not_found_data"][key].keys():
    some_dict["not_found_data"][key]["provider"] = []
some_dict["not_found_data"][key]["provider"].append(provider_name)
```

Maybe it could be beneficial to extract this into a helper function?
```
def add_missing_provider(not_found_data, key, provider_name):
    if "provider" not in not_found_data[key]:
        not_found_data[key]["provider"] = []
    not_found_data[key]["provider"].append(provider_name)
```